student/src/main/java/tds/student/services/ExamCompletionServiceImpl.java

```diff
@@ -44,12 +44,13 @@ public ReviewTestServiceImpl(@Qualifier("integrationOpportunityService") final I
     }
 
     @Override
-    public ResponseData<String> reviewTest(final TestOpportunity testOpportunity, final TestManager testManager)
+    public ResponseData<String> updateStatusWithValidation(final TestOpportunity testOpportunity, final TestManager testManager,
+                                                           final String statusCode)
         throws ReturnStatusException {
         ResponseData<String> responseData = new ResponseData<>(TDSReplyCode.OK.getCode(), "OK", null);
 
         if (isLegacyCallsEnabled) {
-            responseData = legacyReviewTest(testOpportunity, testManager);
+            responseData = legacyReviewTest(testOpportunity, testManager, statusCode);
```

Comment:
Good point, it would set the simply call the updateStatus endpoint on ExamService, which may or may not be successful, depending on what you pass in. I can add code to validate that the status is either "review" or "completed". Also, renaming this private method